flink-cdc-composer/src/main/java/org/apache/flink/cdc/composer/flink/deployment/YarnApplicationDeploymentExecutor.java

```diff
@@ -0,0 +1,132 @@
+        }
+        flinkConfig.set(
+                YarnConfigOptions.SHIP_FILES,
+                additionalJars.stream().map(Path::toString).collect(Collectors.toList()));
+
+        flinkConfig.set(ApplicationConfiguration.APPLICATION_ARGS, commandLine.getArgList());
+        flinkConfig.set(
+                ApplicationConfiguration.APPLICATION_MAIN_CLASS,
+                "org.apache.flink.cdc.cli.CliFrontend");
+        final YarnClusterClientFactory yarnClusterClientFactory = new YarnClusterClientFactory();
+        final YarnClusterDescriptor descriptor =
+                yarnClusterClientFactory.createClusterDescriptor(flinkConfig);
+        ClusterSpecification specification =
+                yarnClusterClientFactory.getClusterSpecification(flinkConfig);
+        ApplicationConfiguration applicationConfiguration =
+                ApplicationConfiguration.fromConfiguration(flinkConfig);
+
+        ClusterClient<ApplicationId> client = null;
+        try {
+            ClusterClientProvider<ApplicationId> clusterClientProvider =
+                    descriptor.deployApplicationCluster(specification, applicationConfiguration);
+            client = clusterClientProvider.getClusterClient();
+            ApplicationId clusterId = client.getClusterId();
+            LOG.info("Deployment Flink CDC From Cluster ID {}", clusterId);
+            return new PipelineExecution.ExecutionInfo(
+                    clusterId.toString(), "submit job successful");
+        } catch (Exception e) {
+            if (client != null) {
+                client.shutDownCluster();
+            }
+            throw new RuntimeException("Failed to deploy Flink CDC job", e);
+        } finally {
+            descriptor.close();
+            if (client != null) {
+                client.close();
+            }
+        }
+    }
+
+    private Optional<String> getFlinkCDCDistJarFromEnv() {
+        String flinkCDCHomeFromEnvVar = System.getenv(FLINK_CDC_HOME_ENV_VAR);
+        Path flinkCDCLibPath = Paths.get(flinkCDCHomeFromEnvVar).resolve("lib");
+        if (!Files.exists(flinkCDCLibPath) || !Files.isDirectory(flinkCDCLibPath)) {
+            LOG.error(
+                    "Flink cdc home lib is not file or not directory: {}",
+                    flinkCDCLibPath.toAbsolutePath());
+            return Optional.empty();
+        }
+        try (Stream<Path> paths = Files.walk(flinkCDCLibPath)) {
+            List<String> distJars = new ArrayList<>();
+            paths.filter(Files::isRegularFile)
+                    .filter(
+                            path ->
+                                    path.getFileName()
+                                            .toString()
+                                            .matches("flink-cdc-dist-.*-.*\\.jar"))
+                    .forEach(path -> distJars.add(String.valueOf(path.toAbsolutePath())));
+            return Optional.ofNullable(distJars.get(0));
```

Comment:
`distJars.get(0)` will never be null. If `distJars` is empty, an `IndexOutOfBoundsException` will be thrown instead of returning `Optional.empty()`.